artiq/frontend/artiq_rtiomap.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+import argparse
+import importlib
+import struct
+
+from sipyco import common_args
+
+from artiq import __version__ as artiq_version
+from artiq.master.databases import DeviceDB
+
+
+def get_argparser():
+    parser = argparse.ArgumentParser(description="ARTIQ RTIO channel name map encoder tool")
+
+    parser.add_argument("--version", action="version",
+                        version="ARTIQ v{}".format(artiq_version),
+                        help="print the ARTIQ version number")
+
+    common_args.verbosity_args(parser)
+    parser.add_argument("--device-db", default="device_db.py",
+                        help="device database file (default: '%(default)s')")
+    parser.add_argument("file", metavar="FILE", default=None,
+                        help="write the result into the specified file")
+
+    return parser
+
+
+def get_rtio_channels(desc):
```

Comment:
Shouldn't all this code be much tighter integrated and shared with [artiq_ddb_template.py](https://github.com/m-labs/artiq/blob/master/artiq/frontend/artiq_ddb_template.py) and [kasli_generic.py](https://github.com/m-labs/artiq/blob/master/artiq/gateware/targets/kasli_generic.py) which both also enumerate RTIO channels and find offsets?

Reply:
I don't see points of integration so far - they seem to do completely different jobs.

Reply:
See the code I linked above. The gateware knows everything about all the RTIO channels, knows there indices and their functionality. You implement RTIO channel enumeration at the coredevice layer all over again. The problems are apparent: duplication and inconsistency.

Reply:
Better doing it in another issue/PR